assets/js/player.js

```diff
@@ -49,6 +49,42 @@ videojs.Vhs.xhr.beforeRequest = function(options) {
 
 var player = videojs('player', options);
 
+player.on('error', () => {
+    if (video_data.params.quality !== 'dash') {
+        if (!player.currentSrc().includes("local=true") && !video_data.local_disabled) {
+            var currentSources = player.currentSources();
+            for (var i = 0; i < currentSources.length; i++) {
+                currentSources[i]["src"] += "&local=true"
+            }
+            player.src(currentSources)
+        }
+        else if (player.error().code === 2 || player.error().code === 4) {
+            setTimeout(function (event) {
```

Comment:
Currently, this error handler retries fetching the video indefinitely. We should probably put a timeout/counter.

Reply:
This is exactly what I described in the "nice to have" list.

I plan to do another PR for that.